### PR TITLE
766 assign listpresence keywords

### PR DESCRIPTION
**Pull request by zachmunzenrider**
Should be easy enough to test. Should show up only for Chemical Presence List datadocuments with associated ExtractedListPresence records. A good one is at http://127.0.0.1:8000/datadocument/254781/

Once you create a comma-separated list of keywords or phrases, you can confirm they exist in the db in ExtractedListPresenceTag 

Conversely, the form shouldn't display at all on http://127.0.0.1:8000/datadocument/7/

### CODE REVIEW

File: dashboard/migrations/0104_auto_20190422_0933.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.AddField(
+            model_name='extractedlistpresence',
+            name='tags',
+            field=taggit.managers.TaggableManager(blank=True, help_text='A set of keywords applicable to this Extracted List Presence', through='dashboard.ExtractedListPresenceToTag', to='dashboard.ExtractedListPresenceTag', verbose_name='Tags'),
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (zachmunzenrider) [reply dashboard/migrations/0104_auto_20190422_0933.py]: Good question! I remember us discussing this in person, but when I looked at the revised developer workflow we made in that same discussion, it makes no mention of handling migrations this way. I'd be on board with changing this, but we should agree and document it

**Comment by zachmunzenrider** [comment]
Matt--it was definitely my understanding from our in-person talk that all listpresence elements should be assigned the same set of keywords. I.e., the relationships exist at the ExtractedListPresence level, but they all have relationships to the same keywords. So I'm fairly confident this implementation matches that requirement.

As to your other comment, about needing to be able to remove all keywords, I agree that this doesn't allow you to do that, but it should! I'm working on that now.

**Comment by mcovalt** [comment]
> Matt--it was definitely my understanding from our in-person talk that all listpresence elements should be assigned the same set of keywords. I.e., the relationships exist at the ExtractedListPresence level, but they all have relationships to the same keywords. So I'm fairly confident this implementation matches that requirement.

Sounds good. That makes sense to me.

> As to your other comment, about needing to be able to remove all keywords, I agree that this doesn't allow you to do that, but it should! I'm working on that now.

Check out that code suggestion I made. That worked for me.

**Comment by zachmunzenrider** [comment]
And that did the trick! Thanks for the suggestion. I've pushed the change up.

**Comment by mcovalt** [comment]
It's a necessary refactor, but I'm curious how the `dashboard/forms.py → dashboard/forms/forms.py` move will be tracked in git.

**Comment by mwfrost** [comment]
I think it means that anyone whose ticket involves `forms.py` should make the `dashboard/forms.py → dashboard/forms/forms.py` move manually before merging `dev` into their working branch.

**Comment by zachmunzenrider** [comment]
Yep. I should've said something about this in my PR notes. Glad you caught it!

File: dashboard/migrations/0104_auto_20190422_0933.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmlyons2) [reply dashboard/migrations/0104_auto_20190422_0933.py]: Need to discuss this at Retro

**Pull request merged**